src/ocm/components/clusterConfiguration/networkConfiguration/StackTypeControl.tsx

```diff
@@ -0,0 +1,130 @@
+import { useFormikContext } from 'formik';
+import { Cluster } from '../../../../common/api/types';
+import { NetworkConfigurationValues } from '../../../../common/types';
+import { NO_SUBNET_SET } from '../../../../common/config/constants';
+import { getFieldId } from '../../../../common/components/ui/formik/utils';
+import { ConfirmationModal, PopoverIcon, RadioField } from '../../../../common/components/ui';
+
+export const StackTypeControlGroup: React.FC<{ clusterId: Cluster['id'] }> = ({ clusterId }) => {
+  const { setFieldValue, values, validateForm } = useFormikContext<NetworkConfigurationValues>();
+  const [openConfirmModal, setConfirmModal] = React.useState(false);
+
+  const setSingleStack = () => {
+    setFieldValue('stackType', 'singleStack');
+    setFieldValue('networkType', 'OpenShiftSDN');
+    setFieldValue('vipDhcpAllocation', true);
+
+    if (values.machineNetworks && values.machineNetworks?.length >= 2) {
+      setFieldValue('machineNetworks', values.machineNetworks.slice(0, 1));
+    }
+    if (values.clusterNetworks && values.clusterNetworks?.length >= 2) {
+      setFieldValue('clusterNetworks', values.clusterNetworks.slice(0, 1));
+    }
+    if (values.serviceNetworks && values.serviceNetworks?.length >= 2) {
+      setFieldValue('serviceNetworks', values.serviceNetworks.slice(0, 1));
+    }
+
+    validateForm();
+  };
+
+  const setDualStack = () => {
+    setFieldValue('stackType', 'dualStack');
+    setFieldValue('networkType', 'OVNKubernetes');
+    setFieldValue('vipDhcpAllocation', false);
+
+    if (values.machineNetworks && values.machineNetworks?.length < 2) {
+      setFieldValue(
+        'machineNetworks',
+        [...values.machineNetworks, { cidr: NO_SUBNET_SET, clusterId: clusterId }],
+        false,
+      );
+    }
+    if (values.clusterNetworks && values.clusterNetworks?.length < 2) {
+      setFieldValue(
+        'clusterNetworks',
+        [...values.clusterNetworks, { cidr: '', hostPrefix: '', clusterId: clusterId }],
+        false,
+      );
+    }
+    if (values.serviceNetworks && values.serviceNetworks?.length < 2) {
+      setFieldValue(
+        'serviceNetworks',
+        [...values.serviceNetworks, { cidr: '', clusterId: clusterId }],
+        false,
+      );
+    }
+  };
+
+  const setStackType = () => {
+    if (values.stackType === 'singleStack') {
+      setDualStack();
+    } else if (
+      (values.machineNetworks && values.machineNetworks[1].cidr !== 'NO_SUBNET_SET') ||
+      (values.clusterNetworks && values.clusterNetworks[1].cidr) ||
+      (values.clusterNetworks && values.clusterNetworks[1].hostPrefix) ||
+      (values.serviceNetworks && values.serviceNetworks[1].cidr)
+    ) {
+      setConfirmModal(true);
+    } else {
+      setSingleStack();
+    }
+  };
+
+  return (
+    <>
+      <FormGroup
+        label="Networking stack type"
+        fieldId={getFieldId('stackType', 'radio')}
+        isInline
+        onChange={setStackType}
```

Comment:
@rawagner explained to me that it works because onChange propogates to the parent if no child stops the propogation. So this is actually a cool way to implement a radio form group